toolkits/microsoft/arcade_microsoft/outlook_calendar/tools/list_events_in_time_range.py

```diff
@@ -0,0 +1,56 @@
+from typing import Annotated
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Microsoft
+from msgraph.generated.users.item.calendar.calendar_view.calendar_view_request_builder import (
+    CalendarViewRequestBuilder,
+)
+
+from arcade_microsoft.client import get_client
+from arcade_microsoft.outlook_calendar._utils import (
+    convert_timezone_to_offset,
+    create_timezone_request_config,
+    get_default_calendar_timezone,
+    replace_timezone_offset,
+)
+from arcade_microsoft.outlook_calendar.models import Event
+
+
+@tool(requires_auth=Microsoft(scopes=["MailboxSettings.Read", "Calendars.ReadBasic"]))
+async def list_events_in_time_range(
+    context: ToolContext,
+    start_date_time: Annotated[
+        str,
+        "The start date and time of the time range, represented in "
+        "ISO 8601 format. Timezone offset is ignored. For example, 2025-04-24T19:00:00",
+    ],
+    end_date_time: Annotated[
+        str,
+        "The end date and time of the time range, represented in "
+        "ISO 8601 format. Timezone offset is ignored. For example, 2025-04-24T19:30:00",
+    ],
+    limit: Annotated[int, "The maximum number of events to return. Max 1000. Defaults to 10"] = 10,
```

Comment:
Based on the [Outlook docs sample response](https://learn.microsoft.com/en-us/graph/api/calendar-list-calendarview?view=graph-rest-1.0&tabs=python#response-1), an event dict has ~300 chars. If a user searches for a long enough period and retrieves 1000 events, the response may potentially contain 300,000 chars, which seems too big.

Perhaps 100 would be more appropriate?

Although the API doesn't offer pagination, we could paginate inside the tool and return `events[offset:offset+limit]`. That would keep response size relatively small, while still allowing to retrieve up to 1,000 events.